tarofreighter/chain_porter.go

```diff
@@ -577,6 +577,14 @@ func adjustFundedPsbt(pkt *tarogarden.FundedPsbt, anchorInputValue int64) {
 // stateStep attempts to step through the state machine to complete a Taro
 // transfer.
 func (p *ChainPorter) stateStep(currentPkg sendPackage) (*sendPackage, error) {
+	// Notify subscribers that the state machine is about to execute a
+	// state.
+	stateEvent := ExecuteSendStateEvent{
```

Comment:
Ahh ok I see the usage here now: we can emit events from the main state machine to relevant RPC callers.